docs/differences-from-v1.md

```diff
@@ -0,0 +1,69 @@
+# Differences from FOSSA 1.x to 2.x
+
+Upgrading from FOSSA CLI 1.x to 2.x is a major breaking change. For most users, this change should be seamless. For users who use less common flags, or who have heavily customized their CLI usage, you may need to update some scripts.
+
+## Table of contents <!-- omit in toc -->
+- [What's new in FOSSA 2.x?](#whats-new-in-fossa-2x)
+  - ["Modules" are now "analysis targets"](#modules-are-now-analysis-targets)
+  - [Automatic analysis target discovery](#automatic-analysis-target-discovery)
+  - [Automatic analysis target strategy selection](#automatic-analysis-target-strategy-selection)
+  - [Improved correctness](#improved-correctness)
+  - [Improved debug logging](#improved-debug-logging)
+- [How to upgrade to FOSSA 2.x](#how-to-upgrade-to-fossa-2x)
```

Comment:
Ideally, we can add a section, on what user should do, if they are facing issue with migration. Might be good idea to have specific [issue template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository) for migration related inquiries/bugs.